zaza/openstack/utilities/openstack.py

```diff
@@ -2740,9 +2740,15 @@ def write_private_key(keypair_name, key):
     :param key: PEM Encoded Private Key
     :type key: str
     """
+    # Create the key file with mode 0o600 to allow the developer to pass it to
+    # the `ssh` command without getting a "bad permissions" error.
+    stored_umask = os.umask(0o177)
+
     with open(get_private_key_file(keypair_name), 'w') as key_file:
         key_file.write(key)
 
+    os.umask(stored_umask)
```

Comment:
this should be part of a try-finally so we always restore the umask no matter if the writing failed.

```
stored_umask = os.umask(0o177)
try:
    with open(get_private_key_file(keypair_name), 'w') as key_file:
        key_file.write(key)
finally:
    os.umask(stored_umask)
```

Reply:
That is definitively more clean, I avoided adding that because the exception raised would have made zaza crash and exit anyway.

But if someone comes in the future to add a except here, this will probably not be thought of and become a latent bug.

I'll add your suggestion. Thx!